Sources/Caching/DeviceCache.swift

```diff
@@ -328,13 +328,38 @@ class DeviceCache {
                                       environment: .init(sandbox: isSandbox))
     }
 
+    // MARK: - Products Entitlements
+
+    var isProductEntitlementMappingCacheStale: Bool {
```

Comment:
cc @aboedo this is done. I made this a constant cache duration instead of depending on `isAppBackgrounded`.